thirdparty/git.go

```diff
@@ -140,6 +153,113 @@ func GetPatchSummaries(patchContent string) ([]patch.Summary, error) {
 	return summaries, nil
 }
 
+func GetPatchSummariesByCommit(mboxPatch string) ([]Summary, []string, error) {
+	commits, err := GetCommitsFromMboxPatch(mboxPatch)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "can't get commits from patch")
+	}
+
+	summaries := []Summary{}
+	commitMessages := []string{}
+	for _, commit := range commits {
+		for i := range commit.Summaries {
+			commit.Summaries[i].Description = commit.Message
+		}
+		summaries = append(summaries, commit.Summaries...)
+		commitMessages = append(commitMessages, commit.Message)
+	}
+
+	return summaries, commitMessages, nil
+}
+
+func GetDiffsFromMboxPatch(patchContent string) (string, error) {
+	commits, err := GetCommitsFromMboxPatch(patchContent)
+	if err != nil {
+		return "", errors.Wrap(err, "problem getting commits from patch")
+	}
+	diffs := make([]string, 0, len(commits))
+	for _, commit := range commits {
+		diffs = append(diffs, commit.Patch)
+	}
+
+	return strings.Join(diffs, "\n"), nil
+}
+
+// GetCommitsFromMboxPatch returns commit information from an mbox patch
+func GetCommitsFromMboxPatch(mboxPatch string) ([]Commit, error) {
+	tmpDir, err := ioutil.TempDir("", "patch_summaries_by_commit")
+	if err != nil {
+		return nil, errors.Wrap(err, "problem creating temporary directory")
+	}
+	defer os.RemoveAll(tmpDir)
+
+	mailSplitCommand := exec.Command("git", "mailsplit", "--keep-cr", fmt.Sprintf("-o%s", tmpDir))
+	mailSplitCommand.Stdin = strings.NewReader(mboxPatch)
+	if err = mailSplitCommand.Run(); err != nil {
+		return nil, errors.Wrap(err, "problem splitting patch content")
+	}
+
+	commits := []Commit{}
+	files, err := ioutil.ReadDir(tmpDir)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem listing split patches")
+	}
+	for i, file := range files {
+		fileReader, err := os.Open(filepath.Join(tmpDir, file.Name()))
+		if err != nil {
+			return nil, errors.Wrap(err, "can't open individual patch file")
+		}
+
+		msgFile := filepath.Join(tmpDir, fmt.Sprintf("%d_message.txt", i))
+		patchFile := filepath.Join(tmpDir, fmt.Sprintf("%d_patch.txt", i))
+		mailInfoCommand := exec.Command("git", "mailinfo", msgFile, patchFile)
+		mailInfoCommand.Stdin = fileReader
+		out, err := mailInfoCommand.CombinedOutput()
+		if err != nil {
+			return nil, errors.Wrap(err, "problem getting mailinfo from patches")
+		}
+		commitMessage, err := parseCommitMessage(string(out))
+		if err != nil {
+			return nil, errors.Wrap(err, "problem parsing commit message")
+		}
+		msgFileInfo, err := os.Stat(msgFile)
+		if err != nil {
+			return nil, errors.Wrap(err, "problem getting message file info")
+		}
+		if msgFileInfo.Size() > 0 {
+			commitMessage = fmt.Sprintf("%s...", commitMessage)
```

Comment:
My understanding is that git breaks up commit messages into a summary and a description. (The message [here](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines) illustrates the two sections).

For the purposes of displaying the commit message on patch pages and commit queue patch descriptions I wanted to exclude the extended description but still indicate to the user that Evergreen knows there's more to their message, so if there's an description section I append an ellipsis to the subject.

Another implementation could be to store the whole message and handle what I wanted to include from the message in each case. I vacillated between these two options for a while, and settled on this one.

Reply:
Ah. Does the extended description still get added when merged? I don't see that message is stored anywhere. If the message isn't part of the commit title when it's merged then I think it should be removed so it doesn't confuse people, otherwise I understand the ellipses choice if it's just for simpler evergreen viewing.

Can you just add a comment for this if statement saying like "indicate to the user that their is more to their commit description" or something?

Reply:
> Does the extended description still get added when merged?

Yes. It's still in the raw patch we supply to `git am` in git.get_project, so `git am` will use the whole message when it recreates the commit. I added a comment included this in the comment as well.